### ISSUE TITLE
JRuby fails on JDK 8u201 on Solaris due to bouncycastle when used via asciidoctor-maven-plugin

### ISSUE THREAD

**Issue by bshannon** [jruby/jruby#5640]
### Environment

* JRuby version 9.2.6.0
* Operating system and platform: Solaris 11.4+ (SunOS datsunx 5.11 11.5.0.38.0 i86pc i386 i86pc)

Using JRuby via the asciidoctor-maven-plugin fails on Solaris with JDK 8u201.  It works on Linux.
It works on on Solaris with JDK 8u191.

### To Reproduce

```
$ mkdir abug
$ cd abug
$ mkdir book
$ touch book/book.adoc
$ cat > pom.xml <<'EOF'
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <groupId>bug</groupId>
    <artifactId>bug</artifactId>
    <packaging>pom</packaging>
    <version>1.0-SNAPSHOT</version>

    <properties>
        <asciidoctor.maven.plugin.version>1.5.7.1</asciidoctor.maven.plugin.version>
        <asciidoctorj.version>1.6.2</asciidoctorj.version>
        <asciidoctorj.pdf.version>1.5.0-alpha.16</asciidoctorj.pdf.version>
        <jruby.version>9.2.6.0</jruby.version>
    </properties>

    <build>
        <defaultGoal>package</defaultGoal>
        <plugins>
            <plugin>
                <groupId>org.asciidoctor</groupId>
                <artifactId>asciidoctor-maven-plugin</artifactId>
                <version>${asciidoctor.maven.plugin.version}</version>
                <dependencies>
                    <dependency>
                        <groupId>org.jruby</groupId>
                        <artifactId>jruby-complete</artifactId>
                        <version>${jruby.version}</version>
                    </dependency>
                    <dependency>
                        <groupId>org.asciidoctor</groupId>
                        <artifactId>asciidoctorj</artifactId>
                        <version>${asciidoctorj.version}</version>
                    </dependency>
                    <dependency>
                        <groupId>org.asciidoctor</groupId>
                        <artifactId>asciidoctorj-pdf</artifactId>
                        <version>${asciidoctorj.pdf.version}</version>
                    </dependency>
                </dependencies>
                <configuration>
                    <sourceDirectory>book</sourceDirectory>
                    <sourceDocumentName>book.adoc</sourceDocumentName>
                    <backend>pdf</backend>
                </configuration>
                <executions>
                    <execution>
                        <id>generate-pdf-doc</id>
                        <phase>generate-resources</phase>
                        <goals>
                                <goal>process-asciidoc</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
        </plugins>
    </build>
</project>
EOF
$ mvn -V
Apache Maven 3.3.3 (7994120775791599e205a5524ec3e0dfe41d4a06; 2015-04-22T04:57:37-07:00)
Maven home: /opt/maven
Java version: 1.8.0_201, vendor: Oracle Corporation
Java home: /usr/jdk/instances/jdk1.8.0/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "sunos", version: "5.11", arch: "amd64", family: "unix"
[INFO] Scanning for projects...
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building bug 1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- asciidoctor-maven-plugin:1.5.7.1:process-asciidoc (generate-pdf-doc) @ bug ---
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] Copying 0 resource
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 11.992 s
[INFO] Finished at: 2019-03-02T20:03:19-08:00
[INFO] Final Memory: 48M/410M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.asciidoctor:asciidoctor-maven-plugin:1.5.7.1:process-asciidoc (generate-pdf-doc) on project bug: Execution generate-pdf-doc of goal org.asciidoctor:asciidoctor-maven-plugin:1.5.7.1:process-asciidoc failed: (NameError) cannot load (ext) (org.jruby.ext.openssl.OpenSSL): InvocationTargetException: class org.bouncycastle.asn1.ASN1Primitive overrides final method equals.(Ljava/lang/Object;)Z -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException

```

See also [this asciidoctor-maven-plugin issue](https://github.com/asciidoctor/asciidoctor-maven-plugin/issues/388).

**Comment by headius** [comment]
@bshannon Thanks for the report! This one escaped my notice.

This looks like there's two versions of BouncyCastle getting used at the same time. We've seen it before.

Here's a StackOverflow question that looks identical: https://stackoverflow.com/questions/17212410/itext-bouncycastle-throws-java-lang-verifyerror-class-overrides-final-method

Did something else change with your setup when updating that JDK? Can you scan for another copy of BouncyCastle somewhere in that environment?

**Comment by bshannon** [comment]
Right you are!!!

The new version of the JDK came with an update to Solaris.
There were old versions of BouncyCastle in /usr/java/jre/lib/ext, perhaps
left over from previous work I did.  I was previously using a separately
installed JDK and when I switched to the version bundled with Solaris
to get the new version of the JDK, these old jre/lib/ext jar files were
now being found and used, causing this problem.

Thanks for the hint, and sorry to bother you.
